sdk/core/azure-core/inc/azure/core/internal/client_options.hpp

```diff
@@ -44,15 +53,15 @@ namespace Azure { namespace Core { namespace Internal {
      */
     explicit ClientOptions(ClientOptions const& options)
```

Comment:
why do we need copy constructor and move constructor?

I suppose this `ClientOptions` won't be used directly. It's only used as base class of other client options. So when are we going to use copy constructor or move constructor?

Reply:
If we don't use it for now, we can remove it and we can always add it later.

Reply:
When an sdk client is created, the clientOptions are copied to the client.  We need the copy constructor to make sure polices are cloned (they are unique_prt).
And the move constructor is to support rvalues for creating a client like
`sdkClient client(url, credential, clientOptions(constructor-for-options));`  we take the rval and move it.

Reply:
I just feel leaving out `RetryOptions`, `TransportOptions` and `TelemetryOptions` in constructors is kind of weird, not intuitive, and can lead to unexpected behaviors.

Reply:
They are not left out. They are members of the clientOptions